src/masternode-payments.cpp

```
@@ -273,12 +285,49 @@ bool CMasternodePayments::IsTransactionValid(const CTransaction& txNew, int nBlo
+            spark::ParseSparkMintTransaction(scripts, mintTransaction);
+        } catch (std::invalid_argument&) {
+            LogPrintf("CMasternodePayments::%s -- ERROR failed to parse mint outputs in block at height %s\n", __func__, nBlockHeight);
+            return false;
+        }
+
+        //checking whether MintTransaction is valid
+        if (!mintTransaction.verify()) {
+            LogPrintf("CMasternodePayments::%s -- ERROR failed to verify mints in block at height %s\n", __func__, nBlockHeight);
+            return false;
+        }
+        mintTransaction.getCoins(coins);
```

cppcheck

[performance] Variable 'type' is assigned in constructor body. Consider performing initialization in initialization list.

(useInitializationList)